### PR TITLE
IGNITE-7877: Improve code style in GA part

### PR DESCRIPTION
**Pull request by techbysample**

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/Chromosome.java [hunk 1]
```diff
@@ -46,7 +48,6 @@
     private long[] genes;
 
     /**
-     *
      * @param genes Primary keys of Genes
      */
     public Chromosome(long[] genes) {
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/Chromosome.java]: Missed javadoc for line 103
```
/** {@inheritDoc} */
    @Override public String toString() {
```

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java [hunk 2]
```diff
@@ -230,10 +231,9 @@ public Chromosome evolve() {
         List<Long> orderChromKeysByFittest = new ArrayList();
         String orderDirection = "desc";
 
-        if(config.isHigherFitnessValueFitter()==false)
-         {
-                orderDirection = "asc";
-         }
+        if (config.isHigherFitnessValueFitter() == false) {
+            orderDirection = "asc";
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java]: braces style and this expression may be improved:
```
if (!config.isHigherFitnessValueFitter())
            orderDirection = "asc";
```

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java [hunk 3]
```diff
@@ -230,10 +231,9 @@ public Chromosome evolve() {
         List<Long> orderChromKeysByFittest = new ArrayList();
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java]: Missed diamond?
`List<Long> orderChromKeysByFittest = new ArrayList<>();`

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/Gene.java [hunk 4]
```diff
@@ -36,6 +36,7 @@
  */
 public class Gene {
 
+    /** primary key of Gene */
     private static final AtomicLong ID_GEN = new AtomicLong();
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/Gene.java]: Missed javadoc for toString()

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateJob.java [hunk 5]
```diff
@@ -42,6 +42,7 @@
     /** primary keys of genes to be used in mutation **/
     private List<Long> mutatedGeneKeys;
 
+    /** Ignite instance */
     @IgniteInstanceResource
     private Ignite ignite = null;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateJob.java]: brace style for line 80

Comment (ybabak) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateJob.java]: And TODO at the line 69

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateTask.java [hunk 6]
```diff
@@ -49,9 +49,11 @@
  */
 public class MutateTask extends ComputeTaskAdapter<List<Long>, Boolean> {
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateTask.java]: empty line here and braces at the lines 145 and 148

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/TruncateSelectionJob.java [hunk 7]
```diff
@@ -42,6 +42,7 @@
     /** primary keys of genes to be used in mutation */
     private List<Long> mutatedGeneKeys;
 
+    /** Ignite instance */
     @IgniteInstanceResource
     private Ignite ignite = null;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/TruncateSelectionJob.java]: TODO at the line 64 and double braces at the line 72

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/TruncateSelectionTask.java [hunk 8]
```diff
@@ -44,9 +44,9 @@
 /**
  * Responsible for performing truncate selection.
  */
-
 public class TruncateSelectionTask extends ComputeTaskAdapter<List<Long>, Boolean> {
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/TruncateSelectionTask.java]: empty line here and please format methods getChromosome and getEnhancedPopulation

Comment (ybabak) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/TruncateSelectionTask.java]: And TODO in reduce method

Comment (techbysample) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/TruncateSelectionTask.java]: ybabak, Please share with me what respective tools you are using to validate code style.  Presently, I am using IntelliJ's format command to apply the respective style using 'ignite_codeStyle.xml'

IE:

format -s <PathToIgniteCodeStyle>\ignite_codeStyle.xml -m *.java -r <PathToIgniteCode>\modules\ml\src\main\java\org\apache\ignite\ml\genetic

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/functions/GAGridFunction.java [hunk 9]
```diff
@@ -35,7 +35,6 @@
  * Responsible for providing custom SQL functions to retrieve optimization results
  */
 public class GAGridFunction {
-
     /** GAConfiguration **/
     private GAConfiguration config = null;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/functions/GAGridFunction.java]: for lines 52 and 63: are those lines is needed?

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/parameter/GAGridConstants.java [hunk 10]
```diff
@@ -34,4 +33,5 @@
     }
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/parameter/GAGridConstants.java]: please cleanup lines 34-36

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/CrossOverJob.java [hunk 11]
```diff
@@ -49,10 +48,11 @@
...
  */
 public class CrossOverJob extends ComputeJobAdapter {
-
+    /** Ignite resource */
     @IgniteInstanceResource
     private Ignite ignite = null;
 
+    /** Ignite logger */
     @LoggerResource
     private IgniteLogger log = null;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/CrossOverJob.java]: please check line 87

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java [hunk 12]
```diff
@@ -79,7 +78,6 @@ public GAGrid(GAConfiguration config, Ignite ignite) {
      *
      * @return Average fitness score
      */
-
     private Double calculateAverageFitness() {
 
         double avgFitnessScore = 0;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java]: Line 105: Do we need return value here?
And the same for line 122, 158, 310, 428

And for 148(+ 404): unnecessary local variable 

Brace style for lines 362 and 365

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateTask.java [hunk 13]
```diff
@@ -67,11 +68,10 @@ public MutateTask(GAConfiguration config) {
...
      */
     private List<Long> getMutatedGenes() {
-        List<Long> mutatedGenes = new ArrayList();
+        List<Long> mutatedGenes = new ArrayList<Long>();
         config.getChromosomeLength();
 
         for (int i = 0; i < config.getChromosomeLength(); i++) {
-            // Gene gene=config.getGenePool().get(selectRandomIndex(config.getGenePool().size()));
             mutatedGenes.add(selectGene(i));
         }
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateTask.java]: Brace style for lines 143 and 146

And for 187: unnecessary local variable?

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/functions/GAGridFunction.java (hunk 9 above)
Comment (techbysample) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/functions/GAGridFunction.java]: Yes.  These line are required.

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/parameter/GAGridConstants.java (hunk 10 above)
Comment (techbysample) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/parameter/GAGridConstants.java]: This is they way source was formatted after applying IntelliJ's format and ignite_codeStyle.xml on source. I have corrected.

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/CrossOverJob.java (hunk 11 above)
Comment (techbysample) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/CrossOverJob.java]: Fixed.

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java (hunk 12 above)
Comment (techbysample) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/GAGrid.java]: Line 105. Fixed
Line 122, 158, 310, 428. Fixed

148(+ 404): - These variable are necessary.
Brace style for lines 362 and 365. Fixed

File: modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateTask.java (hunk 13 above)
Comment (techbysample) [reply modules/ml/src/main/java/org/apache/ignite/ml/genetic/MutateTask.java]: Brace style for lines 143 and 146. Fixed.

And for 187: unnecessary local variable - This variable is necessary.

**Pull request closed without merging**